File: algorithm/src/main/java/baekjoon/p02985/Main.java
package baekjoon.p02985;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt(), sc.nextInt(), sc.nextInt()));
    }

    private static String solve(int a, int b, int c) {
        String oper = oper(a, b, c);
        String oper2 = oper(b, c, a);

        if (oper != null) {
            return String.format("%d%s%d=%d", a, oper, b, c);
        } else {
            return String.format("%d=%d%s%d", a, b, oper2, c);
        }
    }

    private static String oper(int a, int b, int c) {
        if (a + b == c) {
            return "+";
        } else if (a - b == c) {
            return "-";
        } else if (a * b == c) {
            return "*";
        } else if (a / b == c) {
            return "/";
        } else {
            return null;
        }
    }

}

File: algorithm/src/main/java/baekjoon/p02985/Main.kt
package baekjoon.p02985

fun main(args: Array<String>) {
    val (a, b, c) = readLine()!!.split(" ").map { it.toInt() }
    println(solve(a, b, c))
}

private fun solve(a: Int, b: Int, c: Int): String {
    val oper = oper(a, b, c)
    val oper2 = oper(b, c, a)

    return when {
        oper.isNotEmpty() -> "$a$oper$b=$c"
        oper2.isNotEmpty() -> "$a=$b$oper2$c"
        else -> ""
    }
}

private fun oper(a: Int, b: Int, c: Int): String {
    return when (c) {
        a + b -> "+"
        a - b -> "-"
        a * b -> "*"
        a / b -> "/"
        else -> ""
    }
}